src/info/manifest.c

```diff
@@ -383,6 +386,750 @@ manifestNewInternal(void)
+            if (strEqZ(buildData.manifestParentName, MANIFEST_TARGET_PGDATA "/" MANIFEST_TARGET_PGTBLSPC))
+            {
+                THROW_FMT(
+                    LinkExpectedError, "'%s' is not a symlink - " MANIFEST_TARGET_PGTBLSPC " should contain only symlinks",
+                    strPtr(manifestName));
+            }
+
+            // Add path to manifest
+            ManifestPath path =
+            {
+                .name = manifestName,
+                .mode = info->mode,
+                .user = info->user,
+                .group = info->group,
+            };
+
+            manifestPathAdd(buildData.manifest, &path);
+
+            // Skip excluded path content
+            if (buildData.excludeContent != NULL && strLstExists(buildData.excludeContent, manifestName))
+            {
+                LOG_INFO(
+                    "exclude contents of '%s/%s' from backup using '%s/' exclusion", strPtr(buildData.pgPath), strPtr(info->name),
+                    strPtr(strSub(manifestName, sizeof(MANIFEST_TARGET_PGDATA))));
+
+                FUNCTION_TEST_RETURN_VOID();
+                return;
+            }
+
+            // Skip the contents of these paths if they exist in the base path since they won't be reused after recovery
+            if (strEq(buildData.manifestParentName, MANIFEST_TARGET_PGDATA_STR))
+            {
+                if (strEqZ(info->name, PG_PATH_PGDYNSHMEM) && pgVersion >= PG_VERSION_94)
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+
+                if (strEqZ(info->name, PG_PATH_PGNOTIFY))
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+
+                if (strEqZ(info->name, PG_PATH_PGREPLSLOT) && pgVersion >= PG_VERSION_94)
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+
+                if (strEqZ(info->name, PG_PATH_PGSERIAL) && pgVersion >= PG_VERSION_91)
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+
+                if (strEqZ(info->name, PG_PATH_PGSNAPSHOTS) && pgVersion >= PG_VERSION_92)
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+
+                if (strEqZ(info->name, PG_PATH_PGSTATTMP))
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+
+                if (strEqZ(info->name, PG_PATH_PGSUBTRANS))
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+            }
+
+            // Skip the contents of archive_status when online
+            if (buildData.online && strEq(buildData.manifestParentName, buildData.manifestWalName) &&
+                strEqZ(info->name, PG_PATH_ARCHIVE_STATUS))
```

Comment:
I don't see this in the Perl code. In Perl we specifically did not skip it (note !~):

        next if ($bOnline && $strFile =~ (qw{^} . MANIFEST_TARGET_PGDATA . qw{/} . $self->walPath() . '\/') &&
                 $strFile !~ ('^' . MANIFEST_TARGET_PGDATA . qw{/} . $self->walPath() . qw{/} . DB_PATH_ARCHIVESTATUS . '$'));

Reply:
The logic is now split between line 540 and 597.  At line 540 we don't exclude pg_xlog/pg_wal because we later may want to preserve the contents (file at 597) or at least `archive_status`.  We don't have all that info at 540 so we just exclude the contents of `archive_status` when online.